lib/netutil/conn_pool_test.go

```diff
@@ -59,3 +64,64 @@ func testConnPoolStartStop(t *testing.T, name string, ms *metrics.Set) {
 		}
 	}
 }
+
+// TestGetPutDialConnectionPool test the concurrent dial limit of the connection pool.
+// It simulates 256 concurrent goroutine to get new connections, while the concurrent dial limit shouldn't be more than 64.
+// After the first 64 goroutines got connections, the rest 192 goroutines will race for the `select-case`.
+// Only 64 of them have chance to create new connections while the rest will go to the default case and try to obtain
+// used connections.
+//
+// Without extra logging or metrics, it's hard to verify if all the paths are covered. But the coverage report shows it
+// does.
+func TestGetPutDialConnectionPool(t *testing.T) {
+	mockSvr := newMockServer()
+	addr, _ := url.Parse(mockSvr.URL)
+	cp := NewConnPool(metrics.NewSet(), "test-pool", addr.Host, mockHandshake, 1, 5*time.Second, 0)
+
+	concurrency := 256
+	connChan := make(chan *handshake.BufferedConn, concurrency)
+
+	// concurrent create connections
+	for i := 0; i < concurrency; i++ {
+		go func() {
+			conn, err := cp.Get()
+			if err != nil {
+				t.Errorf("get conn from connection pool err:%v", err)
```

Comment:
it should be fatal or return on err. Otherwise, it will put nil to chan, and consumer will successfully read it on L:100

Reply:
updated, thanks!